server/drivers/athena/index.js

```diff
@@ -268,8 +276,29 @@ const fields = [
       'If set, only Athena databases which follow these regex patterns will be shown in the sidebar. ' +
       'The patterns must be separated by commas.',
   },
+  {
+    key: 'maxrows_override',
+    formType: 'TEXT',
+    label: 'Maximum rows to return',
+    description: 'Optional',
+  },
 ];
 
+/**
+ * Check if number is a positive integer, otherwise return default
+ * @param {object} num
+ * @param {number} defaultValue
+ * @returns
+ */
+function resolvePositiveNumber(num, defaultValue) {
+  if (num == null) return defaultValue;
+  if (typeof num === 'string') num = Number.parseInt(num, 10);
+  if (typeof num !== 'number') return defaultValue;
+  if (!Number.isFinite(num)) return defaultValue;
+  if (num > 0) return num;
+  return defaultValue;
+}
```

Comment:
Could this be extracted and put in util/lib directory with some tests?

Reply:
Done!